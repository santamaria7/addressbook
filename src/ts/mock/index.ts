import configureMockStore from "redux-mock-store";
import createSagaMiddleware from "redux-saga";
import rootSaga from "../store/saga";
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
export function mockStore(){
    configure({ adapter: new Adapter() });
    const sagaMiddleware = createSagaMiddleware();
    const mockStoreAPI = configureMockStore([sagaMiddleware]);

    const store = mockStoreAPI({
        users: [],
        pagination: {
            limit: 1000,
            offset: 1,
        },
        search: {
            isFiltered: false,
        },
        loading: false,
        nationality: "",
    });
    sagaMiddleware.run(rootSaga);
    return store
}