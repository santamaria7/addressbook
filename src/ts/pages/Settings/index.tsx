import React from "react";

function Settings () {
  return (
    <div className="page">
        <h1 className="page__title">Settings</h1>
    </div>
  );
};
export default Settings;
