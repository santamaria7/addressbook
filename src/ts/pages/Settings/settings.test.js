import renderer from "react-test-renderer";
import {Provider} from "react-redux";
import {mockStore} from "../../../../mock";
import React from "react";
import Settings from "./index";
import {shallow} from "enzyme";

jest.mock("react-router-dom", () => {
    const originalModule = jest.requireActual("react-router-dom");
    return {
        ...originalModule,
        useHistory: () => ({
            goBack: jest.fn(),
        }),
    };
});

describe("SettingsPage", function () {
    it("should render settings page - with snapshot", function () {
        const component = renderer.create(<Provider store={mockStore()}>
            <Settings />
        </Provider>);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
    it("should render settings page - without snapshot", function () {
        const wrapper = shallow(
            <Provider store={mockStore()}>
                <Settings />
            </Provider>
        );
        expect(wrapper.find("Settings").length).toEqual(1);
    });
});