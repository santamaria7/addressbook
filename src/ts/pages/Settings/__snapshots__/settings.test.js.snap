// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SettingsPage should render settings page - with snapshot 1`] = `
<div
  className="page"
>
  <h1
    className="page__title"
  >
    Settings
  </h1>
  <div
    className="settings"
  >
    <form
      onSubmit={[Function]}
    >
      <label>
        Set preferred nationality:
      </label>
      <select
        onChange={[Function]}
        value=""
      >
        <option
          value="CH"
        >
          CH
        </option>
        <option
          value="ES"
        >
          ES
        </option>
        <option
          value="FR"
        >
          FR
        </option>
        <option
          value="GB"
        >
          GB
        </option>
      </select>
      <button
        type="submit"
      >
        Save
      </button>
    </form>
  </div>
</div>
`;
